tests/BasicRegistryTests.cpp

```diff
@@ -1167,6 +1182,79 @@ TEST_CASE("BasicRegistryTests::wstrings", "[registry]")
         }
     }
 
+#if defined(_VECTOR_)
+    SECTION("get_value_nothrow with non-null-terminated string: with opened key")
+    {
+        wil::unique_hkey hkey;
+        REQUIRE_SUCCEEDED(wil::reg::create_key_nothrow(HKEY_CURRENT_USER, testSubkey, &hkey, wil::reg::key_access::readwrite));
+        REQUIRE_SUCCEEDED(wil::reg::set_value_byte_vector_nothrow(hkey.get(), stringValueName, REG_SZ, nonNullTerminatedString));

```

Comment:
I suggest we don't actually provide any helpers that let you write strings with embedded nulls as values (eg this overthrow that supports a custom `REG_SZ` type, which isn't really aligned with any other of our helpers).

I'd simply call manual registry write calls for the test :D

Simply put, I'm skeptical of the value of allowing folks to pass custom `REG_` types in our helpers, especially since this overload only exists for byte vector.

(Unrelated, did we ever add support for `wstring_view`/`zwstring_view`?)

Reply:
Oh, I see that `byte_vector` can read/write any type, which needs to be specified... the more I think about it, the more it makes sense.